apps/server/src/shared/repo/user/user.repo.integration.spec.ts

```diff
@@ -399,4 +399,26 @@ describe('user repo', () => {
 			expect(user.id).not.toBeNull();
 		});
 	});
+
+	describe('delete', () => {
+		it('should delete user', async () => {
+			// Arrange
+			const user1: User = userFactory.buildWithId();
+			const user2: User = userFactory.buildWithId();
+			const user3: User = userFactory.buildWithId();
+
+			await em.persistAndFlush([user1, user2, user3]);
+			// em.clear();
+
+			await repo.delete(user1);
+			const result1 = await em.find(User, { id: user1.id });
+			expect(result1).toHaveLength(0);
+
+			const result2 = await repo.findById(user2.id);
+			expect(result2).toBeInstanceOf(User);
+
+			const result3 = await repo.findById(user3.id);
+			expect(result3).toBeInstanceOf(User);
```

Comment:
I think you could do the full document check here (to verify that the content didn't change for the non-deleted users) which would be a more complete test.

Reply:
changed